vfs.go

```diff
@@ -603,3 +603,50 @@ func (v *VFSfh) fetchAndSetSize() error {
 
 	return nil
 }
+
+// FolderData is a type encapsulating list of folders and files
+type FolderData struct {
+	Folders []string
+	Files   []string
+	Vfs     *VFS
+}
+
+//export vfsLs
+func vfsLs(path *C.cchar_t, data unsafe.Pointer) int32 {
+	folderData := pointer.Restore(data).(*FolderData)
+
+	uri := C.GoString(path)
+
+	isDir, err := folderData.Vfs.IsDir(uri)
+	if err != nil {
+		return 0
+	}
+
+	if isDir {
+		folderData.Folders = append(folderData.Folders, uri)
+	} else {
+		folderData.Files = append(folderData.Files, uri)
+	}
+
+	return 1
+}
+
+// Ls returns list of folders and files in a path
+func (v *VFS) Ls(path string) ([]string, []string, error) {
```

Comment:
Do we want this to be called `ls` for consistency with the TileDB C code, or do we want this to be named `List`? I am partial to `List` so that the name is less opaque.

Reply:
List works for me too